team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -1865,6 +1871,21 @@ protected ITokenComparator createTokenComparator(String line) {
 		return new TokenComparator(line);
 	}
 
+	/**
+	 * Creates an <code>IIgnoreWhitespaceContributor</code> which allows to hook
+	 * into the ignore whitespace logic in the compare viewer. Tool specific
+	 * implementations can overrule which whitespaces can be ignored and which not
+	 * (e.g. whitespaces in literals).
+	 *
+	 * @param document
+	 * @return a IIgnoreWhitespaceContributor which allows to overrule the platform
+	 *         based whitespace ignore logic in the compare view
+	 * @since 3.9
+	 */
+	protected IIgnoreWhitespaceContributor createIgnoreWhitespaceContributor(IDocument document) {
+		return null;
```

Comment:
Why do we introduce `null` values to the API to then check for it later? Can we have stub implementation for this interface here and never have `null` in the calling code?

Reply:
Or instead a better way instead of `nulls` is using [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) ...

Reply:
I wasn't sure how this hook would be used: is it `optional`, `required` or 'multiple'?

Reply:
could you please take a look at 1997436 ?

Reply:
Much better, and if we use `Optional` we can 
1) require non-null for constructor argument
2) remove `null` check

Reply:
How do you want that the constructor argument is checked for non-null? I tried annotation @NonNull, but didn't work. How is this done today in the eclipse framework code?

Reply:
for example `java.lang.Objects.requireNonNull(T)` in constructor

Reply:
> done with [77496e4](https://github.com/eclipse-platform/eclipse.platform/commit/77496e47798a705fb95d3c9d131a3c5d331f83c5) This was the first occurrence of Objects.requireNonNull call in the compare plugin. Is this really ok?

Why shouldn't it be ok.